src/ElizabethCausalModels/CausalModels.jl

```diff
@@ -0,0 +1,155 @@
+module CausalModels
+
+using Distributions
+using Statistics
+export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+
+# notes for how to do the functions for objects
+# struct ExogenousVariable2
+#   name
+#   stat
+# end
+#
+# courtorder = ExogenousVariable2(:courtorder, Bernoulli(0.5))
+# function (v::ExogenousVariable2)(input)
+#   return input
+# end
+# courtorder(1)
+
+#Returns a function that takes in inputs and returns different values
+#based on whether use_stat, prob, or get_name are true
+#use_stat returns a random value from the distribution'
+#prob returns the name and mean to be use for probability calculations
+#get_name returns the symbolic name of the var
+#otherwise uses the given variable substitution or returns random value
+function ExogenousVariable(name, stat)
+  function insert_tuples(tuple, use_stat=false, prob=false, get_name=false)
+    if get_name
+      return name
+    end
+    if use_stat
+      return rand(stat)
+    end
+    if prob
+      return (name, mean(stat))
+    end
+    if haskey(tuple, name)
+      tuple[name]
+    else
+      rand(stat)
+    end
```

Comment:
This function is doing several different kinds of things.  instead of putting all these clauses inside insert_tuple, you should use different methods applied to different types.